### DISCUSSION TITLE
SCI / SCER values for CNCF projects [Green Reviews Working Group]

### DISCUSSION THREAD

**Discussion by ArneTR** [cncf/tag-env-sustainability#273] [General]
Hey All,

I found my way to this repo through the links in the GSF Newsletter. Happy to see that the CNCF has an environmental sustainability groups :)

I am Arne, one of the creators of the free and open source [Green Metrics Tool](https://github.com/green-coding-berlin/green-metrics-tool/) (Thanks for the mention by the way on your [Landscape page](https://tag-env-sustainability.cncf.io/landscape/) ❤️  )

Last week I had a chat with @mkorbi and talked about ideas to make software more sustainable and about your efforts to create a snapshot of the current emissions / sustainability of all your CNCF products via the [Green Reviews WG](https://github.com/cncf/tag-env-sustainability/blob/main/working-groups/green-reviews/charter.md)

We have been working on something similar for arbitrary software with our Green Metrics Tool, which creates [SCI scores](https://docs.green-coding.berlin/docs/measuring/sci/) in an automated fashion and then also displays that data nicely in [some statistical Dashboards](https://www.metrics.green-coding.berlin) as well as an [Energy-ID scorecard](https://www.green-coding.berlin/projects/energy-id/)

Here an example where [we worked with the Wagtail community](https://github.com/wagtail/wagtail/discussions/8843) to create a energy and carbon consumption snapshot as well as an SCI score for their reference implementation [The Wagtail Bakery](https://github.com/green-coding-berlin/bakerydemo-gold-benchmark)

- [Link to the Energy-ID "scorecard"](https://www.green-coding.berlin/projects/energy-id/) (picture attached)
- [Link to the timeline of historical energy changes](https://metrics.green-coding.berlin/timeline.html?uri=https://github.com/green-coding-berlin/bakerydemo-gold-benchmark&filename=usage_scenario_warm.yml&branch=&machine_id=7)
- [Link to a sample measurement with SCI score, badges etc.](https://metrics.green-coding.berlin/stats.html?id=79b0ef85-19dd-4f16-b519-48f4c92c5092) where we monitor the project with every commit 
- [Link of the comparison functionality of the Green Metrics Tool with a prior commit of the Wagtail project](https://metrics.green-coding.berlin/compare.html?ids=473e4f5a-986c-4a0e-8683-9fa13e6a5cd3,cfda6233-7011-46d5-a692-e5ad301ff4d2) Screenshot attached

<img width="391" alt="Screenshot 2023-11-14 at 10 29 26 AM" src="https://github.com/cncf/tag-env-sustainability/assets/250671/94a4b2e8-30a8-4706-8f1b-ce700f9b4f17">
Energy-ID Scorecard

<img width="2244" alt="Screenshot 2023-11-14 at 10 39 33 AM" src="https://github.com/cncf/tag-env-sustainability/assets/250671/d8c42b50-a019-4c05-84d9-459131c64232">
Comparison functionality

## GSF

We also proposed this automated mechanism to the GSF here and they [launched their SCER standard](https://github.com/Green-Software-Foundation/scer/pull/7), which tries to provide a reference framework for tackling the task that your are doing.
- [Link to SCER discussion about automating with the Green Metrics Tool](https://github.com/Green-Software-Foundation/scer/issues/1)
- [Link to SCI discussion about automating with the Green Metrics Tool](https://github.com/Green-Software-Foundation/sci-guide/issues/52)

## Summary

I wanted to open this discussion particular to ask if we can if the Green Metrics Tool and it's capabilities would be in any form a help for your venture in the Green Reviews WG and am happy to help if so.

**Comment by leonardpahlke** [comment]
Hi Arne, this looks interesting! Thanks for reaching out. I will have a closer look later this week. 

cc @nikimanoledaki  @catblade

**Reply by nikimanoledaki** [reply]
Thanks for the ping @leonardpahlke! 👍

Hi @ArneTR! 👋  Would you be interested in doing a short demo (~5-10 min) of the Green Metrics Tool during one of the upcoming TAG ENV meetings followed by a Q&A discussion? In total we should try to stay under 30-40 min to leave some time for TAG updates etc.

For the demo, I suggest emphasizing equally how the tool was built and implemented from a platform perspective as well as how it's used from a user perspective. The platform side would help the WG Green Reviews since we are creating a similar platform with/for cloud-native tooling. 

I propose the [meeting on the 17th of January](https://calendar.google.com/calendar/u/0/embed?src=72e93a411f02e5664bb4485c04311b83dae6a62574e4ab882a1ccf8526aa9bf1@group.calendar.google.com&ctz=America/Chicago) when most folks are back from PTO.

It would be great to learn more about how you identify the different factors of the SCI. Personally, what I would find the most interesting is if you could share any pointers on how to narrow down the functional unit of a tool. This part can be more challenging since "how it scales" can be qualitative/subjective. Identifying and implementing this is not always straightforward with cloud-native tooling. We could even take 5-10 min to identify a couple of cloud-native tooling and come up with the functional unit together. This would be a great group exercise. :)

**Reply by ArneTR** [reply]
Hey @nikimanoledaki  , thanks for the follow-up on this.

Sure, happy to do a presentation. On the 17th I however have a long train trip during the day, so I am only available in the evening CST time. What time will the meeting be?

**Reply by nikimanoledaki** [reply]
Hi @ArneTR, the meeting is at 5pm CET. The next meeting is in 2 weeks on the 31st of January. Let me know which one works for you!

**Reply by ArneTR** [reply]
Hey @nikimanoledaki . Oh, that is too bad ... I will not be available then. Shall we fix the 31st then? This one is free in my calendar. Same time 5 pm?

**Reply by nikimanoledaki** [reply]
Wonderful, let's do that! I'll share it in today's TAG ENV meeting 👍

**Reply by nikimanoledaki** [reply]
@ArneTR really looking forward to discussing the Green Metrics Tool in tomorrow's meeting! Please let me know if there is anything that I can do to help.
Meeting details can be found [here](https://calendar.google.com/calendar/u/0/embed?src=72e93a411f02e5664bb4485c04311b83dae6a62574e4ab882a1ccf8526aa9bf1@group.calendar.google.com&ctz=America/Chicago)

**Reply by ArneTR** [reply]
thanks for the ping @nikimanoledaki . I have it penciled in and looking forward to later.

**Reply by nikimanoledaki** [reply]
Wonderful! See you at 5pm CET - links:
- Agenda & Minutes: https://bit.ly/cncf-tag-env-meeting-notes
- Zoom Link: https://zoom.us/my/cncftagenvsustainability, passcode 77777

**Reply by ArneTR** [reply]
Hey @leonardpahlke @nikimanoledaki ,

thanks for opportunity last week regarding the presentation. I was sadly sick right after and wanted to follow up on this.

We talked about Falco and your current efforts of measuring / baselining it atm and if we replicate that with the Green Metrics Tool from our side. Is this still current? If so, can you send me some details for it? We are very keen on doing some measurements on it :)

**Reply by nikimanoledaki** [reply]
Hi @ArneTR! A few notes from last week's Green Reviews WG meeting where we discussed this (see [meeting notes](https://docs.google.com/document/d/19xUur8RjnNwO2rwLw5Wwf7ISQ9V41sr0UaartIJoQvE/edit#heading=h.ldw7pr4je486)):
- We'll be able to pick this up until after KubeCon (end of March) since we have a tight deadline until then.
- We can learn from this project especially to learn how to do some of the things we have no plan for e.g. embedded carbon metrics 
- We would really appreciate your input and expertise if you would like to join WG Green Review meetings or [other comm channels](https://github.com/cncf-tags/green-reviews-tooling?tab=readme-ov-file#community)

Also, @by-d-sign raised that we could integrate the Eco-CI tool
- Issue: https://github.com/cncf-tags/green-reviews-tooling/issues/55
- We'll revisit this after KubeCon :)

**Reply by ArneTR** [reply]
@nikimanoledaki Thanks for the follow up on this!

I do understand the KubeCon timing issues.

What we can do from our side is create a mirror and then a PR for some of your unit tests. I will follow up in the linked issue.

Ping me when you folks are ready for the Falco test :)

2. I think I cannot attend the regular Green Review meetings since my calendar is already exploding. But am happy to attend on a case by case basis. Especially when it is tooling related. Ping me and I will try to be there.
